bin/utils.sh

```diff
@@ -31,25 +31,30 @@ download_url() {
     version="$2"
   fi
 
-  case "$(uname -s)" in
-    "Linux")
-      platform=linux64
-      ;;
-    "Darwin")
-      if [ "$version" = "nightly" ]; then
-        case "$(uname -p)" in
-          "arm")
-            platform=macos-arm64
-            ;;
-          *)
-            platform=macos-x86_64
-            ;;
-        esac
-      else
-        platform=macos
-      fi
-      ;;
-  esac
+	case "$(uname -s)" in
+	"Linux")
+		platform=linux64
+		;;
+	"Darwin")
+		# Check version is less than v0.10.0 - if it is, use the old platform name
+		# using IFS to split the version number into an array
+		IFS='.' read -r -a version_array <<<"${version#v}"
+
+		# version_array[0] is the major version and version_array[1] is the minor version
+		if ((version_array[0] == 0 && version_array[1] < 10)) && [[ ! $version =~ ^stable|nightly$ ]]; then
+			platform=macos
+		else
+			case "$(uname -p)" in
+			"arm")
+				platform=macos-arm64
+				;;
+			*)
+				platform=macos-x86_64
+				;;
+			esac
+		fi
+		;;
+	esac
```

Comment:
Thank you very much! Could you please fix the indentation here? Looks like the case has an extra level 😅

Reply:
Nevermind, this is blocking people. I'll update it later